tests/unit/string_lib_test.c

```diff
@@ -1507,6 +1507,80 @@ static void test_StringJoin(void)
     SeqDestroy(seq);
 }
 
+static void test_StringSplit(void)
+{
+    {
+        Seq *const seq = StringSplit("foo", ",");
+        assert_int_equal(SeqLength(seq), 1);
+        assert_string_equal(SeqAt(seq, 0), "foo");
+        SeqDestroy(seq);
+    }
+    {
+        Seq *const seq = StringSplit("foo,bar", ",");
+        assert_int_equal(SeqLength(seq), 2);
+        assert_string_equal(SeqAt(seq, 0), "foo");
+        assert_string_equal(SeqAt(seq, 1), "bar");
+        SeqDestroy(seq);
+    }
+    {
+        Seq *const seq = StringSplit("foo,bar,baz", ",");
```

Comment:
Add a test with a trailing delimiter (and fix the bug in the code), please.

Reply:
There is a test below with a trailing delimiter
```C
    {
        Seq *const seq = StringSplit("/foo//bar\\/baz\\", "\\/");
        assert_int_equal(SeqLength(seq), 7);
        assert_string_equal(SeqAt(seq, 0), "");
        assert_string_equal(SeqAt(seq, 1), "foo");
        assert_string_equal(SeqAt(seq, 2), "");
        assert_string_equal(SeqAt(seq, 3), "bar");
        assert_string_equal(SeqAt(seq, 4), "");
        assert_string_equal(SeqAt(seq, 5), "baz");
        assert_string_equal(SeqAt(seq, 6), "");
        SeqDestroy(seq);
    }
```
what bug are you referring to ?

Reply:
OK, so the empty string at the end of the sequence is a feature. I see.